zerver/lib/api_test_helpers.py

```diff
@@ -929,6 +953,7 @@ def test_streams(client, nonadmin_client):
     toggle_mute_topic(client)
     update_subscription_settings(client)
     get_stream_topics(client, 1)
+    delete_stream(client, 1)
```

Comment:
What is happening is that you are deleting the `Denmark` stream (which has the ID 1). But `test_messages` below uses that stream, so when you run the tests `test_messages` can't find the `Denmark` stream.

Here's a solution: go change `get_stream_id` to return the ID of the stream and then pass that ID to `delete_stream`. That should fix it.

Here's the diff:
``` diff
diff --git a/zerver/lib/api_test_helpers.py b/zerver/lib/api_test_helpers.py
index 9908d5a8d..757531260 100644
--- a/zerver/lib/api_test_helpers.py
+++ b/zerver/lib/api_test_helpers.py
@@ -202,7 +202,7 @@ def get_profile(client):
                          check_if_exists=check_if_exists)
 
 def get_stream_id(client):
-    # type: (Client) -> None
+    # type: (Client) -> int
 
     # {code_example|start}
     # Get the ID of a given stream
@@ -212,6 +212,8 @@ def get_stream_id(client):
 
     validate_against_openapi_schema(result, '/get_stream_id', 'get', '200')
 
+    return result['stream_id']
+
 def delete_stream(client, stream_id):
     # type: (Client, int) -> None
 
@@ -476,6 +478,8 @@ def send_message(client):
     result = client.send_message(request)
     # {code_example|end}
 
+    print(result)
+
     validate_against_openapi_schema(result, '/messages', 'post', '200')
 
     # test that the message was actually sent
@@ -946,14 +950,14 @@ def test_streams(client, nonadmin_client):
     add_subscriptions(client)
     test_add_subscriptions_already_subscribed(client)
     list_subscriptions(client)
-    get_stream_id(client)
+    stream_id = get_stream_id(client)
     get_streams(client)
     get_subscribers(client)
     remove_subscriptions(client)
     toggle_mute_topic(client)
     update_subscription_settings(client)
     get_stream_topics(client, 1)
-    delete_stream(client, 1)
+    delete_stream(client, stream_id)
 
     test_user_not_authorized_error(nonadmin_client)
     test_authorization_errors_fatal(client, nonadmin_client)

```

Reply:
Yeah. I just checked and it worked. Actually, I tried this before as well. But, I was doing few things wrong like I called both delete_stream and then get_stream_topics with the same ID obtained. 

Nonetheless, it worked. I feel that I should have debugged it. 
I'll complete adding the other API endpoints and update the PR.